src/diamond/handler/cloudwatch.py

```diff
@@ -170,6 +181,11 @@ def process(self, metric):
         metricname = str(metric.getMetricPath())
         timestamp = datetime.datetime.fromtimestamp(metric.timestamp)
 
+        if self.autoscaling:
+            instance = self.ec2.get_only_instances([self.instance_id])
+            self.log.debug(instance[0].tags['aws:autoscaling:groupName'])
+            autoscaling_group = instance[0].tags['aws:autoscaling:groupName']
```

Comment:
Thanks for your comments! Good thought, I think the error should occur, but it should be a more obvious error that tells the user why the key doesn't exist. If they're trying to report to autoscaling on an instance that is not in autoscaling, then we should tell them thats what they've done.